activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -310,6 +310,11 @@ def execute_simple_calculation(operation, column_name, distinct) #:nodoc:
         type_cast_calculated_value(result.cast_values.first, operation) do |value|
           type = column.try(:type_caster) ||
             lookup_cast_type_from_join_dependencies(column_name.to_s) || Type.default_value
+
+          if operation == "average" && type.is_a?(Type::Integer)
+            type = Type::Decimal.new
+          end
```

Comment:
How about `to_d` like as before? Looks like it is overkill that instantiating `Type::Decimal` and using `deserialize` to me.

```suggestion
          if operation == "average" && type.is_a?(Type::Integer)
            value.to_d
          else
            type.deserialize(value)
          end
```

Reply:
I'm not into the matter enough to decide wether we can take this shortcut here:

But my argument against the suggested change would be that `ActiveModel::Type::Decimal#cast_value` contains a lot of (I guess justified) logic which determines the correct kind of `BigDecimal` to return. I imagine that database adapters could return anything while `type` still is a `Type::Integer`. With your suggestion you weight performance over stability, correct?

Reply:
The Decimal type is generally used by `attribute :foo, :decimal`, the `foo` attribute will be passed various types of values, the purpose of the `cast_value` is to accept the values and handle the precision and the scale.

In this case, `value` is a numeric or a string or nil, that is why I say it is overkill.

Reply:
Ok. I took in your suggestion.